src/analysis/complex/upper_half_plane.lean

```diff
@@ -76,79 +78,156 @@ begin
   fin_cases i; assumption,
 end
 
-lemma denom_ne_zero (g : SL(2, ℝ)) (z : ℍ) : denom g z ≠ 0 :=
-linear_ne_zero (g 1) z (g.row_ne_zero 1)
+lemma det_of_22 {R : Type*} [comm_ring R] (M: matrix (fin 2) (fin 2) R):
+  M.det = (M 0 0) * (M 1 1) - (M 0 1) * (M 1 0):=
+begin
+rw  matrix.det_succ_row_zero,
+simp [fin.sum_univ_succ],
+ring,
+end
```

Comment:
This exists already, it's `matrix.det_fin_two`.

Reply:
Oh great! thanks.